build: re-enable deployment-time acceptance tests for tag-based builds

See #5697 

They are currently failing with:
```
time acceptance/acceptance.test -test.v -test.timeout 5m -i cockroachdb/cockroach -nodes 3 -l "${CIRCLE_ARTIFACTS}"/acceptance_deploy 2>&1 > "${CIRCLE_ARTIFACTS}/acceptance_deploy.log"
sys 0m0.008s
I160329 23:11:03.006152 rand.go:86  Random seed: -8532244621296312315
I160329 23:11:03.093272 acceptance/cluster/localcluster.go:279  Initializing Cluster AdHoc 3x1:
{"name":"AdHoc 3x1","nodes":[{"count":3,"stores":[{"count":1,"max_ranges":0}]}],"duration":5000000000}
I160329 23:11:03.096110 acceptance/cluster/localcluster.go:639  stopping
panic: Error: No such image: cockroachdb/cockroach [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x755a9d]
```

--- a/circle.yml
+++ b/circle.yml
@@ -53,7 +53,20 @@ deployment:
   release:
     tag: /beta-[0-9]+/
     commands:
+      - sed "s/<EMAIL>/$DOCKER_EMAIL/;s/<AUTH>/$DOCKER_AUTH/" < "resource/deploy_templates/.dockercfg.template" > ~/.dockercfg
+      - |
+          export VERSION=$CIRCLE_TAG
+          echo "Deploying ${VERSION}..."
+          if [ -n "$DOCKER_EMAIL" ]; then
+            build/push-docker-deploy.sh
+          fi
       - aws configure set region us-east-1
       - build/build-static-binaries.sh
+      - mkdir -p "${CIRCLE_ARTIFACTS}/acceptance_deploy"
+      - time acceptance/acceptance.test -test.v -test.timeout 5m
+          -i cockroachdb/cockroach -nodes 3
+          -l "${CIRCLE_ARTIFACTS}"/acceptance_deploy 2>&1 >
+          "${CIRCLE_ARTIFACTS}/acceptance_deploy.log"
+
       - build/build-osx.sh
       - build/push-tagged-aws.sh

